samples/CMakeLists.txt

```diff
@@ -5,6 +5,43 @@
 #
 
 if (CONFIG_SPM)
+  if (CONFIG_BUILD_S1_VARIANT AND NOT CONFIG_BOOTLOADER_MCUBOOT)
```

Comment:
nothing wrong as such.
Just started to wonder if this is the right location for spm related CMake code ?

If user creates an out-of-tree app, then this code is still executed, but doesn't seem right to load `ncs/nrf/sample` code into oot app.

Similar when building a Zephyr sample for nrf9160 or nrf53, then one also want to use SPM.

Reply:
I agree that this is not an optimal place to have this code, but I struggle to find something better.

The problem is that we are checking the configuration of the parent image, not SPM itself, so we cannot move this to `samples/spm/CMakeLists.txt`. 

We should maybe try to come up with a new cmake file for all child image logic that we do here.